### DISCUSSION TITLE
lsp.log shows error entries but almost everything seems to work (-code completion)

### DISCUSSION THREAD

**Discussion by ambushed** [clangd/clangd#1294] [Q&A] [answered]
I am using Neovim 7.0 with clangd 13.0 LSP on MacOS for C++ development.  After realizing my code completion doesn't work i had a look into `lsp.log `and realized it was full of errors, while lines logged with errors didn't seem to point to any particular issue (to my inexperienced eye):
[lsp.log](https://github.com/clangd/clangd/files/9578795/lsp.log)
`:LspInfo` shows clangd being attached:
![Screenshot 2022-09-15 at 22 09 06](https://user-images.githubusercontent.com/361353/190502661-86a8a8f1-e869-40b8-a6a7-dbf9809c17d0.png)
all of this is while i can jump to definitions, get code diagnostics and can rename variables.  The only thing is that my code completion doesn't work.   Here is the my lua lsp config for nvim:
[lsp-config-lua.txt](https://github.com/clangd/clangd/files/9578818/lsp-config-lua.txt)

It would be nice to know why i see `ERROR` next to what should really be info messages and maybe that will lead me to fixing my issues with code completion... 

Any help or pointers will be appreciated!

Vlad

**Comment by sam-mccall** [comment, marked as answer]
I don't see any evidence of clangd errors there.
lsp.log is a neovim log file and those `[ERROR]`s at the start of each line are written by neovim, I'm not sure why they do that.

As for completion: there are no code completion requests in that log (i.e. the client is not asking clangd for completions). Neovim LSP doesn't support any completion out of the box, you can [set up omnifunc completion](https://github.com/neovim/nvim-lspconfig#suggested-configuration) (manually triggered) but you're probably looking for [autocompletion](https://github.com/neovim/nvim-lspconfig/wiki/Autocompletion) e.g. nvim-cmp.

**Reply by ambushed** [reply]
Thank you, i will ask a question about ERROR status of info messages to nvim lsp client crowd
